endpoints/openrtb2/sample-requests/invalid-whole/no-site-or-app.json

```diff
@@ -19,5 +19,5 @@
     ]
   },
   "expectedReturnCode": 400,
-  "expectedErrorMessage": "Invalid request: request.site or request.app must be defined, but not both.\n"
+  "expectedErrorMessage": "Invalid request: request.site should include at least one of request.site.id or request.site.page.\n"
```

Comment:
Why did this error message change?

Reply:
@hhhjort 
As per old code, setSiteImplicitly was not setting the site object even if its nil.
Hence, https://github.com/ashishshinde-pubm/prebid-server/blob/Issue_3052/endpoints/openrtb2/auction.go#L1724 function always use to return from here.
But now, since we are explicitly setting the site object inside `setSiteImplicitly` hence we will never get the site object as nil and it will return from line-https://github.com/ashishshinde-pubm/prebid-server/blob/Issue_3052/endpoints/openrtb2/auction.go#L1728